devito/ir/iet/nodes.py

```diff
@@ -521,6 +534,31 @@ def write(self):
         return []
 
 
+class While(Node):
+
+    """
+    Implement a while-loop.
+
+    Parameters
+    ----------
+    condition : sympy.Function or sympy.Relation or bool
+        The while-loop exit condition.
+    body : Node or list of Node, optional
+        The whie-loop body.
+    """
+
+    is_While = True
+
+    _traversable = ['body']
+
+    def __init__(self, condition, body=None):
+        self.condition = condition
+        self.body = as_tuple(body)
```

Comment:
private _?

Reply:
in theory yes, in practice no, because none of the IET classes use private attribute

it's a massive rewrite which we don't wanna do now, and perhaps never?

Reply:
Oh ok, for later maybe then, was just saying to have consistent private over the codebase but I get it's tidious